warehouse/upload.go

```diff
@@ -1067,11 +1068,31 @@ func (job *UploadJobT) loadTable(tName string) (alteredSchema bool, err error) {
 		job.recordTableLoad(tName, numEvents)
 	}
 
-	if columnThreshold, ok := columnCountThresholds[job.warehouse.Type]; ok {
-		columnCount := len(job.schemaHandle.schemaInWarehouse[tName])
-		if columnCount > columnThreshold {
-			job.counterStat(`warehouse_load_table_column_count`, tag{name: "tableName", value: strings.ToLower(tName)}).Count(columnCount)
+	job.columnCountStat(tName)
+
+	return
+}
+
+func (job *UploadJobT) columnCountStat(tableName string) {
+	var (
+		columnCountLimit int
+		ok               bool
+	)
+	if columnCountLimit, ok = columnCountLimitMap[job.warehouse.Type]; !ok {
+		return
+	}
+	currentColumnsCount := len(job.schemaHandle.schemaInWarehouse[tableName])
+	if currentColumnsCount > int(float64(columnCountLimit)*columnCountLimitThreshold) {
+		tags := []tag{
+			{
+				name: "tableName", value: strings.ToLower(tableName),
+			},
+			{
+				name: "columnCountLimit", value: strconv.Itoa(columnCountLimit),
+			},
 		}
+
```

Comment:
Would it make sense to report two metrics? 

`warehouse_load_table_column_count`

`warehouse_load_table_column_limit`

And instead of computing the threshold here, we compute it on the alert manager, by doing a similar operation

Reply:
Should we add other identifiers as well, like workspaceId and destinationId ?

Reply:
Can you elaborate the motivation behind computing it at the alert manager rather than here ? @lvrach

Reply:
> Should we add other identifiers as well, like workspaceId and destinationId ?

We are adding these inside `job.counterStat(`warehouse_load_table_column_count`, tags...).Count(currentColumnsCount)`

Reply:
`warehouse_load_table_column_limit` This is basically a constant. We should not probably send it as a stat.

Reply:
> Can you elaborate the motivation behind computing it at the alert manager rather than here? @lvrach

Since the threshold dictates whether an alert should be sent, it is more natural for me to have it on the alert manager. It makes it easier to change it. Also, the fact that we add `tableName` & `columnCountLimit` only if the threshold passes prevents us from doing back-testing (going to grafana dashboard and checking when the alert will be triggered with another threshold). 

> warehouse_load_table_column_limit This is a constant. We should probably not send it as a stat.
If you want to visualise the limit as a line in grafana it is easier if it's metric.

Reply:
> Also, the fact that we add tableName & columnCountLimit only if the threshold passes prevents us from doing back-testing (going to grafana dashboard and checking when the alert will be triggered with another threshold).

Good point.

Reply:
> `warehouse_load_table_column_limit` This is a constant. We should probably not send it as a stat.
> 
> If you want to visualise the limit as a line in grafana it is easier if it's metric.

I still feel this information is not much useful. Recording in separate stat and adding tags warehouseId, tableName for an almost constant value (changes only on destination type) would be tedious. Adds computational cost to kapacitor or prometheus executor(future) too.
wdyt?

Reply:
Better for debugging and visualization.